### ISSUE TITLE
Client Credential grant_type

### ISSUE THREAD

**Issue by kvslee** [simov/grant#19]
Great awesome library! Does Grant-Express handle Client Credential grant type (without authorization code). If so, can I find how to set it up? 

In config.json, I set it up:
```javascript
  "custom": {
 	"access_url": "https://myapitprovider/token",
	"oauth": 2,
	"key": "xxxxx",
	"secret": "xxxxxx",
	"callback": "/handle_callback"
  }
```
Thanks.

**Comment by simov** [comment]
Currently Grant supports only 3-legged server side flow. You can use the [request](https://github.com/request/request) module for `grant_type: 'client_credentials'`. Here is example for Twitter:

```js
var request = require('request')
request.post('https://api.twitter.com/oauth2/token', {
  form: {
    client_id: '...',
    client_secret: '...',
    grant_type: 'client_credentials'
  },
  json: true
}, function (err, res, body) {/*body contains the credentials*/})
```

Let me know if it works for you.

**Comment by kvslee** [comment]
Thanks for your quick answer. I could successfully receive access_token with Request. 

Simeon, I find that you are the author of Purest, which also uses request.  So, when I use Purest, I can pass the access_code like this.

```javascript
customProvider.query()
  .select('endpoint')
  .auth('access-token')
  .request(function (err, res, body) {});
```

**Comment by simov** [comment]
Yep, this is how you pass your credentials using the query API in Purest.

Let me know if you have any problems. Also I'll think about adding this type of flow. Closing this issue for now.

**Comment by kvslee** [comment]
Yes, adding 'client credentials' grant type would be awesome. I like your library because it structures API providers with endpoints. API request codes looks much cleaner.

**Comment by simov** [comment]
@kvslee I think adding support for `grant_type:client_credentials` will introduce security issue with the current implementation. If you are going to use this type of flow on your web server, you want to have some logic to ensure that the user is authorized to make such request.

Besides that I changed a few things in Purest (currently not on NPM), and actually I introduced a breaking change regarding the `refresh` method, which is no longer available. Take a look at the examples [here](https://github.com/simov/purest#oauth) on how to request various grant types.

Let me know what do you think.

**Comment by kvslee** [comment]
@simov I did not realize that I could use Purest to do client_credentials.Thanks for the example. And it is also nice to see that you added refresh_token grant type. I like the fact that oauth endpoint can be defined in the provider configuration. Are you going to publish the updates into NPM soon?

**Comment by simov** [comment]
Yes I'm going to release the new version soon, in about 2 days at most.

**Comment by simov** [comment]
@kvslee I just released version 1.2.0 of Purest, here is the [changelog](https://github.com/simov/purest/blob/master/CHANGELOG.md).

If you have any other issues regarding Purest, address them there.

Thanks for the feedback :+1:
